modules/core/src/main/scala/lucuma/odb/api/model/targetModel/SourceProfileModel.scala

```diff
@@ -296,45 +296,53 @@ object SourceProfileModel {
       (sed.notMissingAndThen("sed")(_.toUnnormalizedSed),
-       brightnesses.notMissingAndThen("brightnesses")(_.traverse(_.create))
-      ).mapN { (sed, bright) =>
-        BandNormalized(sed, SortedMap.from(bright.map(_.toTuple)))
-      }
-
-    override val edit: StateT[EitherInput, BandNormalized[T], Unit] = {
-      val validArgs = (
-        sed.validateNotNullable("sed")(_.toUnnormalizedSed),
-        brightnesses.validateIsNotNull("brightnesses")
-      ).tupled.toEither
-
+       brightnesses.notMissingAndThen("brightnesses")(brightnessCreator[T]).andThen { m =>
+         // It would be weird to specify edits to the brightnesses you just
+         // created, but it seems like it should work if you do.  Hence:
+         editBrightnesses
+           .toOption
+           .fold(m.validNec[InputError])(brightnessEditor[T](_).runS(m).toValidated)
+       }.map { m =>
+         // Also weird to delete bands you just made, but also should work.
+         deleteBrightnesses.toOption.fold(m)(bands => m -- bands)
+       }
+      ).mapN { (sed, bright) => BandNormalized(sed, bright) }
+
+    override val edit: StateT[EitherInput, BandNormalized[T], Unit] =
       for {
-        args  <- StateT.liftF(validArgs)
-        (s, b) = args
-        _     <- BandNormalized.sed[T]          := s
-        _     <- BandNormalized.brightnesses[T] :< b.map { lst =>
-                   StateT.modifyF[EitherInput, SortedMap[Band, BrightnessMeasure[T]]] { m =>
-
-                     // When editing, the `brightnesses` input is taken to mean
-                     // the list of edits to perform.  Any band not mentioned is
-                     // left unchanged.
-                     val updates: EitherInput[List[(Band, Of[Measure[BrightnessValue], Brightness[T]])]] =
-                       lst.traverse { bbi =>
-                         m.get(bbi.band).fold(bbi.create.map(_.toTuple)) { measure =>
-                           bbi.edit.runS(BandBrightnessPair[T](bbi.band, measure)).map(_.toTuple).toValidated
-                         }
-                       }.toEither
-
-                     updates.map(m ++ _)
-                   }
-                 }
+        s <- StateT.liftF(sed.validateNotNullable("sed")(_.toUnnormalizedSed).toEither)
+        _ <- BandNormalized.sed[T]          := s
+
+        // `brightnesses` to set brightness values if you want to start over
+        // from scratch
+        _ <- BandNormalized.brightnesses[T] :<
+               brightnesses.fold(
+                 brightnessNoop[T],
+                 StateT.setF(emptyBrightnessMap[T].rightNec[InputError]),
+                 in => StateT.setF(brightnessCreator[T](in).toEither)
+               ).some
+
+        // `editBrightnesses` to edit existing ones or add new ones
+        _ <- BandNormalized.brightnesses[T] :<
+               editBrightnesses.toOption.fold(brightnessNoop[T])(brightnessEditor[T]).some
+
+        // `deleteBrightnesses` to delete existing ones
```

Comment:
Maybe we need a `deleteAll`?

Reply:
That might be easier.  I thought about it but you can also accomplish that with setting it to an empty list via `"brightnesses": []`